### DISCUSSION TITLE
Limit How Many Times Location Can Be Changed

### DISCUSSION THREAD

**Discussion by ajbott** [PullDating/Org#147] [Ideas & Features]
Implement a control that only allows people to change their location every X days.

This feature could prevent people from continuously being marked as new to town.

**Comment by conaman551** [comment]
Maybe better to lock location to gps?

**Comment by viktor-lesyk** [comment]
Yet some people can travel every day from one city to another. 
Is "new in town" something that necessary?

Plus I agree, location access fixes the issue with people cheating with their location

**Reply by flo-kei** [reply]
Location mocking might become an issue if we opt for simply using the user's location
(e.g. on android it's quite easy to pretend to be somewhere else using apps available on the Play-Store and ticking a box in the developer options. It should be noted that I have no idea if this can be detected/circumvented)

Another app, Jodel, for local communities lets users set their home-town to their location and if they want to change it, they can do so in the settings, but only once every month. 

I feel like limiting the amount of times a person can change this might be they way to go and fine for 95% of users.
Of course there are folks traveling a lot (and maybe also want to meet people there). 
We would have to figure out how to handle this, but for a first prototype edge-cases like this shouldn't bother us

**Reply by Oshyan** [reply]
Location handling can be tricky. In general I am in favor of using transparency and user options (e.g. filtering) to allow people to make up their own mind about what they show and what they see, rather than imposing hard limits. Generally speaking this is what makes sense to me:

- Consider a "home" vs. "traveling/visiting" location (i.e. two locations)
- Have a "soft" limit on number of changes for "home"; once that limit is reached the user gets a friendly "warning" that they should use Travel Mode if they frequently change locations temporarily and that changing Home more often will be indicated to potential matches who may filter them out
- Indicate *very* clearly whether someone is "home" or "traveling"
- Allow users when browsing/searching to filter on A: anyone in "Travel mode" and B: anyone who changed their Home location more than X times in the last 30 days (or something like that, specifics to be determined)
- Potentially users could either filter *out* those things or filter to *only* those things, at least for Travel Mode (i.e. they only want to see people traveling)

What this hopefully does is allow people who have a legitimate need to change their location frequently to do so, encourages them to not abuse that ability, and empowers other users to know when that is the case and to optionally choose not to see people who are behaving in ways they find questionable/concerning (i.e. changing location frequently).

**Reply by flo-kei** [reply]
I Love this idea!

I mean we need to be careful not to overload our filters (people should still be able to use them easily), but I guess figuring that out is another discussion.

You made me realise that traveling people are certainly something we should incorporate

**Reply by Oshyan** [reply]
Yes, designing the filter UI/UX is its own issue. I really want to provide lots of info, transparency, and options for seeing who you want, which implies a lot of options, UI elements and affordances, etc. So avoiding complexity will have some work to it. 😅 But I am hopeful that many of these things can be solved in relatively simple ways, for example a relatively small set of filters that are basic, and then an entire, expandable "advanced filter" section. Anyone who wants to filter on more than the basics presumably cares enough to wade through many additional options if necessary. Ideally there would also be a super quick "find filter" feature similar to how e.g. Android Phone Settings can be Searched by name for particular settings (this also allows the benefit of accounting for synonyms and descriptive words that match to a filter that someone might not otherwise think to look for).

Similarly a "Simple" vs "Complete" profile view option, where the Simple view would be more like what most dating apps have now, and the "Complete" would show literally all info, like perhaps message response rate, frequency of changing location, etc. etc. So e.g. just because someone wants to *filter out* people who change location a lot doesn't necessarily mean they want to *see location changes indicated on each person's profile*. Those should, I think, be separate possibilities. But I don't think it's worth for example providing a fully customizable profile view where you can turn on/off each individual piece of profile info. Basic vs. Full/Complete seems adequate to me.

Bottom line is I think the average person is going to be wanting a streamlined interface and will accept some loss of control to have that. But there is a subset of people who want total control and/or information and are willing to accept an increase in complexity and information density on-screen in exchange for that, even if those people might *ideally* want some subset of the extra information/complexity. They will *accept* and be happy with "all info/complexity" vs. "too simple and I feel like I can't adjust things well enough". I'm one of those latter people who wants the full control and info density. 😁

**Reply by ajbott** [reply]
**Overview**:
- This feature request aims to introduce a dual-location system within the app to distinguish between a user's permanent "home" location and temporary "traveling/visiting" locations. Additionally, it proposes a soft limit on how often the "home" location can be changed, along with advanced filtering options for users based on location preferences.

**Detailed Requirements**:
- Dual-Location System Implementation:
- Integrate a system that allows users to set and manage two distinct location types: a permanent "home" location and a temporary "traveling/visiting" location.
- Ensure the user interface clearly differentiates between these two location types for ease of understanding and interaction.

**Soft Limit on Home Location Changes:**
- Implement a soft limit on the number of times a user can change their "home" location within a specified time frame (e.g., X changes per 30 days).
- Once a user reaches this limit, display a friendly notification advising them to switch to "Travel Mode" for temporary location changes to avoid frequent "home" location updates.
- Inform users that exceeding the "home" location change limit may affect their visibility, as potential matches can filter based on frequency of location changes.

**Clear Indication of Current Location Status:**
- Design the user interface to clearly indicate whether a user's current location is set as "home" or "traveling/visiting."
- This distinction should be visible both on the user's profile and to other users who view the profile.

**Advanced Filtering Options for Browsing/Searching**:
- Develop advanced filtering capabilities that allow users to refine their browsing/searching based on others' location status.
- Include filter options for:
  a. Users currently in "Travel Mode" (i.e., those temporarily visiting a location different from their home).
  b. Users who have changed their "home" location more than a predetermined number of times in the last 30 days.
- Provide users with the flexibility to either include or exclude profiles based on these criteria, catering to different preferences and needs.

**User Control and Preferences**:
- Ensure that users have control over their location settings and can easily switch between "home" and "traveling" status.
- Allow users to customize their filtering preferences to either see only travelers, only those who frequently change their "home" location, or a combination of both, among other options.

**Objective**:
- The goal of these enhancements is to provide a more flexible, transparent, and user-friendly location management system within the app, fostering a trustworthy environment that accommodates the dynamic lifestyles of our user base while preventing potential misuse of location settings.

[AJ Notes]
- I'd like to see recommended/suggested filters based on location changes rather than having to set a location change. 
 - I think the following would be some good options ["Recently Updated (<30 days)", "Hasn't Changed", "Multichange (low)", "Multichange (medium)", "Multichange (high)"]

**Reply by Oshyan** [reply]
> I'd like to see recommended/suggested filters based on location changes rather than having to set a location change.

Can you clarify what you mean by this?

> I think the following would be some good options ["Recently Updated (<30 days)", "Hasn't Changed", "Multichange (low)", "Multichange (medium)", "Multichange (high)"]

Functionally those might be decent options, but the wording is confusing IMO. Maybe instead of "multichange", something "Changes: Never, Infrequently, Occasionally, Frequently". That - or something similar - would better unify the language and clearly communicate the general frequency of changes.
